java/ql/lib/semmle/code/java/security/TemplateInjectionQuery.qll

```diff
@@ -0,0 +1,40 @@
+/** Provides a taint tracking configuration for server-side template injection (SST) vulnerabilities */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.security.TemplateInjection
+
+/** A taint tracking configuration to reason about server-side template injection (SST) vulnerabilities */
+class TemplateInjectionFlowConfig extends TaintTracking::Configuration {
+  TemplateInjectionFlowConfig() { this = "TemplateInjectionFlowConfig" }
+
+  override predicate isSource(DataFlow::Node source) { this.isSource(source, _) }
```

Comment:
Should probably remove this line (and the corresponding `isSink`). What this currently means is that any source with a non-empty flow state will also be a source with an empty flow state, which probably wasn't intended.

Reply:
I think removing them makes the most sense (your code snippet with `this.isSource(source, any(DataFlow::FlowStateEmpty s))` ought to be equivalent to removing them).